feat(gatsby-plugin-offline): Allow appending custom scripts to sw.js (#11626)
gatsbyjs/gatsby: packages/gatsby-plugin-offline/README.md

plugins: [`gatsby-plugin-offline`]
```

## Available options

As of `gatsby-plugin-offline` 2.1.0, the following options are available:

- `workboxConfig` allows you to override the default Workbox options - see [Overriding Workbox configuration](#overriding-workbox-configuration)

> - `injectScript` lets you specify a file to be injected into the end of the generated service worker (`sw.js`)

Reviewer: Could you add an example here? Something that demonstrates why you might use this option.

For example:

```javascript
plugins: [
  {
    resolve: `gatsby-plugin-offline`,
    options: {
      workboxConfig: {
        importWorkboxFrom: `cdn`,
      },
      injectScript: `src/custom-sw-code.js`,
    },
  },
]
```

## Upgrading from 2.0.x

> To upgrade from 2.0.x to 2.1.x, move any existing options into the `workboxConfig` option. If you haven't specified any options, you have nothing to do.

Reviewer 2: we would need to do major version bump for changing shape of plugin options (it's breaking change)

plugins: [`gatsby-plugin-offline`]
```

## Available options

As of `gatsby-plugin-offline` 3.0.0, the following options are available:

> - `appendScript` lets you specify a file to be injected into the end of the generated service worker (`sw.js`). For example:

Reviewer 3: ```suggestion
- `appendScript` lets you specify a file to be appended at the end of the generated service worker (`sw.js`). For example:
```
